RocketChat/Rocket.Chat.ReactNative
# [IMPROVE] Check permission to create a room

## Proposed changes

* Check if the user has permission to create each one inside the `NewMessageView`: Channel, Team, Direct Message, and Discussion
* In `RoomListView`check if the user has the permissions to create, and if has at least one permission show the button to create, else don't show the button
* In `CreateChannelView` check which permission the user has: if there is permission to create a public channel, a private channel, or both.

We're showing create team, channel, etc without checking if the user has the correct permission.
We should look for it first.

## Issue(s)	



## How to test or reproduce


You need to be the Admin of server to test correctly this, so:

- 1st Create another account as User role and logs in from mobile

- then with the Admin's account go to `Administration > Permissions`, disable the permissions for user: `Create Public Channel`, `Create Private Channel`, `Create Team`, `Create Direct Messages`, `Start Discussion`
 -- In `RoomListView` the mobile's user shouldn't see the button to create a room when all those permissions are disabled for him. To show the button needs at least one of those permissions available.

- back to the Admin's account, you can disable/enable those permissions
 -- In NewMessageView for each permission you will see the types of rooms that you can create. To unauthorized the user to create a channel, the Admin's should disable `Create Public Channel`, `Create Private Channel` for him.

- Depend on the user's permission `Create Public Channel`, `Create Private Channel` will appear differently to create a channel.
* In `CreateChannelView` you will able to see if you can create a private, public channel, or both.

## Screenshots
<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</thead>
</tr>
<body>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976626-f11aed80-d36a-11eb-8b85-26d9b1033646.png" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976693-098b0800-d36b-11eb-9540-a47f40119676.png"  height="450"/>
</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976093-63d79900-d36a-11eb-8ebb-0ee63058fea4.png" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976208-82d62b00-d36a-11eb-91f0-80629a3db449.png"  height="450"/>
</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976777-232c4f80-d36b-11eb-9b49-c5d3ff87024e.png" height="450" />
</td>
<td>
<img src="https://user-images.githubusercontent.com/47038980/122976865-39d2a680-d36b-11eb-8300-21fb4ec2bc8b.png"  height="450"/>
</td>
</tr>
</tbody>
</table>

## Types of changes




## Checklist



## Further comments
